specify/specify7
# Ensure disciplines are resolved correctly when setting uniqueness rules

Fixes #4430 

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions
In a Collection which has a Numeric Auto-Incrementing Formatter 
- Go a new Collection Object form 
- Save the Collection Object with the `#########` formatted catalogNumber
- Ensure the Collection Object saves and increments properly
